src/earn/EarnEnterAmount.tsx

```diff
@@ -144,6 +144,7 @@ function EarnEnterAmount({ route }: Props) {
   }
 
   const {
+    // @ts-expect-error prepareTransactionsResult & swapTransaction are only set when mode === 'deposit' or 'swap-deposit'
```

Comment:
Again, the ts-expects-error is a big red flag. The underlying types should probably be changed or we just need separate functions